src/views/blockDetails/blockBody/index.tsx

```diff
@@ -0,0 +1,236 @@
+    setModalData(data);
+    toggleVisibility();
+  };
+
+  const handleFilter = (e: React.MouseEvent<HTMLButtonElement>) => {
+    const filterType = e.currentTarget.getAttribute('data-filter-type') as 'extrinsics' | 'events';
+    if (filterType) {
+      setFilter(filterType);
+      setShowMore(false);
+    }
+  };
+
+  const toggleShowMore = () => {
+    setShowMore(!showMore);
+  };
+
+  const visibleExtrinsics = showMore ? bodyData.extrinsics : bodyData.extrinsics.slice(0, 2);
+  const visibleEvents = showMore ? bodyData.events : bodyData.events.slice(0, 3);
+
+  const isMoreContentAvailable =
+    (filter === 'extrinsics' && bodyData.extrinsics.length > 2)
+    || (filter === 'events' && bodyData.events.length > 3);
+
+  return (
+    <div className="grid gap-4">
+      <div className="flex gap-6 px-2">
+        <Button
+          isActive={filter === 'extrinsics'}
+          type="extrinsics"
+          onClick={handleFilter}
+        >
+          Extrinsics
+        </Button>
+        <Button
+          isActive={filter === 'events'}
+          type="events"
+          onClick={handleFilter}
+        >
+          Events
+        </Button>
+      </div>
+      {filter === 'extrinsics' && (
+        <div className="overflow-x-auto">
+          <table className="w-full">
+            <colgroup>
+              <col className="min-w-28" />
+              <col className="min-w-24" />
+              <col className="min-w-24" />
+              <col className="min-w-20" />
+              <col className="min-w-24" />
+              <col className="min-w-12" />
+            </colgroup>
+            <thead>
+              <tr className={cn(
+                'bg-dev-purple-100 text-left',
+                'dark:bg-dev-black-900',
+              )}
+              >
+                <th className="py-4 pl-2 font-geist font-body2-bold">Extrinsic ID</th>
+                <th className="py-4 pl-2 font-geist font-body2-bold">Height</th>
+                <th className="py-4 pl-2 font-geist font-body2-bold">Time</th>
+                <th className="py-4 pl-2 font-geist font-body2-bold">Result</th>
+                <th className="py-4 pl-2 font-geist font-body2-bold">Action</th>
+                <th className="py-4 pl-2 font-geist font-body2-bold" />
+              </tr>
+            </thead>
+            <tbody>
+              {visibleExtrinsics.map((extrinsic: IMappedBlockExtrinsic) => {
+                const timeAgo = extrinsic.timestamp && formatDistanceToNowStrict(
+                  new Date(extrinsic.timestamp),
+                  { addSuffix: true },
+                );
+                return (
+                  <tr
+                    key={extrinsic.id}
+                    className={cn(
+                      'pd-table-row',
+                      ' text-dev-purple-600 dark:text-dev-purple-550',
+                    )}
+                    onClick={() => handleOpenModal(extrinsic)}
```

Comment:
no! ... we already have visibleEvents ... we need only the index here and useCallback